src/plugins/vis_type_table/public/table_vis_type.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { CoreStart, PluginInitializerContext } from 'kibana/public';
+import { i18n } from '@kbn/i18n';
+import { AggGroupNames } from '../../data/public';
+import { Schemas } from '../../vis_default_editor/public';
+import { Vis } from '../../visualizations/public';
+import { tableVisResponseHandler } from './table_vis_response_handler';
+// @ts-ignore
+import tableVisTemplate from './table_vis.html';
+import { TableOptions } from './components/table_vis_options';
+import { getTableVisualizationControllerClass } from './vis_controller';
+
+export function getTableVisTypeDefinition(core: CoreStart, context: PluginInitializerContext) {
+  return {
+    type: 'table',
+    name: 'table',
+    title: i18n.translate('visTypeTable.tableVisTitle', {
+      defaultMessage: 'Data Table',
+    }),
+    icon: 'visTable',
+    description: i18n.translate('visTypeTable.tableVisDescription', {
+      defaultMessage: 'Display values in a table',
+    }),
+    visualization: getTableVisualizationControllerClass(core, context),
```

Comment:
Woa, good catch, I completely missed it directly imported the start contract in the old version - super helpful. It seems a bit dangerous though to move the visualization registration into `start` here: https://github.com/elastic/kibana/pull/63105/files#diff-381e25468de5bf2e485a1957d63c9b07R62

I followed the call stack and it seems this is just necessary because of `initLocalAngular` here: https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/vis_type_table/public/vis_controller.ts#L57

But the local angular is only initialized in the `render` function (https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/vis_type_table/public/vis_controller.ts#L63), so we can do the registration in the setup phase, pass down the `CoreSetup` contract and then in `render` get `CoreStart` via `await coreSetup.getStartServices()`. This way we are doing everything in the "right" phase

Reply:
many thx @flash1293 , adapted it that way